addons/partnership/tests/test_partnership.py

```diff
@@ -21,6 +21,41 @@ def test_sell_basic_partnership(self):
             "Selling the partnership should assign the pricelist to the partner",
         )
 
+    def test_sell_basic_partnership_with_children_partners(self):
+        self.partner_with_children = self.env['res.partner'].create({
```

Comment:
We don't store in self local variables only used in this test
```suggestion
        partner_with_children = self.env['res.partner'].create({
```